rosidl_adapter/test/test_constant.py

```diff
@@ -30,6 +30,16 @@ def test_constant_constructor():
     with pytest.raises(ValueError):
         Constant('bool', 'FOO', None)
 
+    # NaN is case insensitive in python, so test a few variations.
+    value = Constant('float32', 'FOO', 'nan')
+    value = Constant('float32', 'FOO', 'Nan')
+    value = Constant('float32', 'FOO', 'NaN')
+    value = Constant('float32', 'FOO', 'NAN')
+    value = Constant('float64', 'FOO', 'nan')
+    value = Constant('float64', 'FOO', 'Nan')
+    value = Constant('float64', 'FOO', 'NaN')
+    value = Constant('float64', 'FOO', 'NAN')
+
```

Comment:
I might be missing something, but this just tests that the calls don't error, right? Maybe this instead?

```suggestion
    # NaN is case insensitive in python, so test a few variations.
    for nan_string in ['nan','Nan','NaN','NAN','nan','Nan','NaN','NAN']:
        value = Constant('float32', 'FOO', nan_string)
        assert isnan(value.value)

```